tests/runtimes/android/android_app_gradle_tests.py

```diff
@@ -0,0 +1,235 @@
+        target_js = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'App_Resources', 'Android', 'app.gradle')
+        File.copy(src=source_js, target=target_js)
+        Tns.platform_remove(app_name=APP_NAME, platform=Platform.ANDROID)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        log = Tns.run_android(APP_NAME, device=self.emulator.id, wait=False, verify=False)
+
+        strings = ['Project successfully built',
+                   'Successfully installed on device with identifier', self.emulator.id,
+                   'Successfully synced application',
+                   '###TEST IMPLEMENTATION PASSED###']
+
+
+        test_result = Wait.until(lambda: all(string in File.read(log.log_file) for string in strings), timeout=300,
+                                 period=5)
+        assert test_result, 'Native packages could not be used with implementation in app.gradle'
+
+    def test_307_native_package_with_api_app_gradle(self):
+        """
+         Test that native packages could be used with api in app.gradle
+         https://github.com/NativeScript/android-runtime/issues/993
+        """
+
+        # Change main-page.js so it contains only logging information
+        source_js = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-993', "api",
+                                 'main-page.js')
+        target_js = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'main-page.js')
+        File.copy(src=source_js, target=target_js)
+        # Change app app.gradle so it contains the dependencies to com.github.myinnos:AwesomeImagePicker:1.0.2
+        source_js = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-993', "api",
+                                 'app.gradle')
+        target_js = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'App_Resources', 'Android', 'app.gradle')
+        File.copy(src=source_js, target=target_js)
+        Tns.platform_remove(app_name=APP_NAME, platform=Platform.ANDROID)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        log = Tns.run_android(APP_NAME, device=self.emulator.id, wait=False, verify=False)
+
+        strings = ['Project successfully built',
+                   'Successfully installed on device with identifier', self.emulator.id,
+                   'Successfully synced application', '###TEST API PASSED###']
+
+        test_result = Wait.until(lambda: all(string in File.read(log.log_file) for string in strings), timeout=300,
+                                 period=5)
+        assert test_result, 'Native packages could not be used with api in app.gradle'
+
+    def test_319_build_project_with_foursquare_android_oauth(self):
+        # This is required when build with different SDK
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(APP_NAME, template=Template.HELLO_WORLD_JS.local_package)
```

Comment:
Can we use Template.MIN_JS?

Reply:
All tests are based on hello world js template. I would prefer using it.